asserts/asserts.go

```diff
@@ -1399,3 +1407,86 @@ func SignatureCheck(assert Assertion, pubKey PublicKey) error {
 	}
 	return nil
 }
+
+// VerifySignature checks the signature of a given boady against the given public key.
+// It is used by model service to verify nonce of the request-id.
+// It currently supports key with algorithms RSA, DSA, ECDSA, and ED25519.
+// All data is expected to use the SHA3-384 hash.
+func VerifySignature(nonce, signature []byte, pubKey crypto.PublicKey) error {
+	switch keyType := pubKey.(type) {
+	case *rsa.PublicKey:
+		return verifyRSAKey(nonce, signature, pubKey.(*rsa.PublicKey))
+	case *dsa.PublicKey:
+		return verifyDSAKey(nonce, signature, pubKey.(*dsa.PublicKey))
+	case *ecdsa.PublicKey:
+		return verifyECDSAKey(nonce, signature, pubKey.(*ecdsa.PublicKey))
+	case ed25519.PublicKey:
+		return verifyED25519Key(nonce, signature, pubKey.(ed25519.PublicKey))
+	default:
+		return fmt.Errorf("unsupported algorithm type: %s", keyType)
+	}
+}
+
+func verifyRSAKey(nonce, signature []byte, pubKey *rsa.PublicKey) error {
+	hash := sha3.New384()
```

Comment:
is there agreement with field to use this as hash for the nonce? is it in the specs?